TCK Tracking: Jakarta EE 10 Core Profile #6799

jbescos opened the pull request:
Relates to https://github.com/helidon-io/helidon/issues/6799

Some TCK tests are excluded in restfull and cdi. You can see them with a `<!-- FIXME: Next tests must be enabled -->` in the pom file. I will create a new issue to address that.

URL of the TCKS:
https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/promoted/eftl/

jbescos commented:
This PR is ready for review.

tomas-langer commented on dependencies/pom.xml:
```diff
@@ -39,6 +39,7 @@
         <version.lib.activemq>5.16.0</version.lib.activemq>
         <version.lib.animal-sniffer>1.18</version.lib.animal-sniffer>
         <version.lib.annotation-api>1.3.5</version.lib.annotation-api>
+        <version.lib.ant>1.10.13</version.lib.ant>
```
Please manage this dependency in top level pom, as this is not for our customers to be used (this file is only for dependencies required at runtime)

jbescos replied:
With `top level pom` do you mean in `helidon/microprofile/tests/tck/pom.xml` or  `helidon/pom.xml`?

jbescos replied:
It was a dead code. I have removed it.

Initially I added it to execute the sigtest from a build.xml of tck-annotations, but it failed because the java ant task uses the security manager and that is not supported.

tomas-langer commented on microprofile/tests/tck/pom.xml:
```diff
@@ -43,6 +43,13 @@
+        <module>tck-jsonb</module>
+        <module>tck-jsonp</module>
+        <module>tck-inject</module>
+	 <module>tck-annotations</module>
```
wrong alignment (line 52)

jbescos replied:
Done

tomas-langer commented on microprofile/tests/tck/tck-cdi/src/test/java/org/jboss/cdi/tck/tests/policy/broken/incorrect/name/stereotype/Mock.java:
```diff
@@ -0,0 +1,31 @@
+@Stereotype
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Mock {}
```
I think this requires some documentation - why does this file even exist?

jbescos replied:
I recall there was a ClassNotFoundException or something like that without it.

I will test again, and I will try to change visibility to `package`

jbescos replied:
Not needed anymore.

tomas-langer commented on microprofile/tests/tck/tck-restfull/artifact-install.sh:
```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2023 Oracle and/or its affiliates. All rights reserved.
+#
+# This program and the accompanying materials are made available under the
```
This is not a correct copyright for Helidon sources

jbescos replied:
I guess I took it from somewhere in Helidon. I have taken other sh file as an example and updated the copyrights.

tomas-langer commented on microprofile/tests/tck/tck-restfull/tck-restfull-test/src/test/resources/logging-test.properties:
```diff
@@ -0,0 +1,24 @@
+java.util.logging.ConsoleHandler.formatter=java.util.logging.SimpleFormatter
+java.util.logging.SimpleFormatter.format=%1$tH:%1$tM:%1$tS %4$s %3$s %5$s%6$s%n
+# Global logging level. Can be overridden by specific loggers
+.level=INFO
```
Please remove unused configuration from git (keep only `root.level` and remove other `.level` properties

jbescos replied:
I have removed it

tomas-langer reviewed (changes requested):
There are a few `.sh` files in the PR, with no explanation in a README.md or simillar - do we need those files? If we do, why and how should we use them?
Any shell script is a danger for us - we need to maintain them, and if called from aynwhere in maven (or ant that was added), it may do things that are unexpected.

jbescos commented:
> There are a few `.sh` files in the PR, with no explanation in a README.md or simillar - do we need those files? If we do, why and how should we use them? Any shell script is a danger for us - we need to maintain them, and if called from aynwhere in maven (or ant that was added), it may do things that are unexpected.

These sh and ant scripts exists because TCK artifacts don't exist in any maven repository. The way it works is that they submit it in a zip file with the jars, and sometimes there is an SH file to install dependencies.
In some zip files the SH file is missing. To be consistent with the other ZIPs, I created the missing SH scripts.

I think there is no need for READMEs because nobody is supposed to understand and execute that. They are executed as a part of the maven ant plugin.

jbescos commented on dependencies/pom.xml:
```diff
@@ -105,6 +106,9 @@
         <version.lib.micronaut.sql>4.4.0</version.lib.micronaut.sql>
         <!-- FIXME upgrade to 3.1 when it is released in Maven -->
         <version.lib.microprofile-config>3.0.1</version.lib.microprofile-config>
+        <version.lib.microprofile-core-profile>10.0.1</version.lib.microprofile-core-profile>
```
Moved to helidon/pom.xml

jbescos commented:
I added the readmes for the sh files

tomas-langer commented on dependencies/pom.xml:
```diff
@@ -105,6 +105,8 @@
         <version.lib.micronaut.sql>4.4.0</version.lib.micronaut.sql>
         <!-- FIXME upgrade to 3.1 when it is released in Maven -->
         <version.lib.microprofile-config>3.0.1</version.lib.microprofile-config>
+        <version.lib.microprofile-cdi-tck>4.0.10</version.lib.microprofile-cdi-tck>
```
TCK dependency still here, should be in root `pom.xml`

jbescos replied:
My bad, I moved only one before. Now every TCK property and dependency is moved to `helidon/pom.xml`

tomas-langer commented on dependencies/pom.xml:
```diff
@@ -75,7 +75,7 @@
         <version.lib.jakarta.jms-api>3.0.0</version.lib.jakarta.jms-api>
         <version.lib.jakarta.jsonb-api>3.0.0</version.lib.jakarta.jsonb-api>
-        <version.lib.jakarta.jsonp-api>2.1.1</version.lib.jakarta.jsonp-api>
+        <version.lib.jakarta.jsonp-api>2.1.0</version.lib.jakarta.jsonp-api>
```
Why are we downgrading this dependency?

tomas-langer commented on dependencies/pom.xml:
```diff
@@ -686,6 +688,21 @@
                     </exclusion>
                 </exclusions>
             </dependency>
+            <dependency>
```
These dependencies should not be defined in this file.

jbescos replied:
Done

tomas-langer commented on dependencies/pom.xml:
```diff
@@ -955,6 +972,11 @@
                     </exclusion>
                 </exclusions>
             </dependency>
+            <dependency>
```
Same here, this dependency should not be here

jbescos replied:
Done

jbescos commented:
TCK Inject failed because I changed the visibility of `microprofile/tests/tck/tck-inject/tck-inject-test/src/test/java/io/helidon/microprofile/tests/inject/AtInjectTCKExtension.java` from `public` to `package`.

I have deleted that class and loaded it from the one that exists in `jakarta.inject-tck-2.0.2-bin.zip`
